### PR TITLE
add sequence id field to ConnectionDataMessage

### PR DESCRIPTION
**Pull request by terencefan**

### CODE REVIEW

File: src/Microsoft.Azure.SignalR.Protocols/ConnectionMessage.cs [hunk 1]
```diff
@@ -136,5 +136,10 @@ public ConnectionDataMessage(string connectionId, ReadOnlySequence<byte> payload
         /// Gets or sets the binary payload.
         /// </summary>
         public ReadOnlySequence<byte> Payload { get; set; }
+
+        /// <summary>
+        /// Gets or sets the sequence id.
+        /// </summary>
+        public int SeqId { get; set; } = 0;
```
Comment (vicancy) [inline src/Microsoft.Azure.SignalR.Protocols/ConnectionMessage.cs]: nit: naming: use full name `SequenceId`?

File: src/Microsoft.Azure.SignalR.Protocols/ServiceProtocol.cs [hunk 2]
```diff
@@ -526,12 +527,15 @@ private static CloseConnectionMessage CreateCloseConnectionMessage(ref MessagePa
             return new CloseConnectionMessage(connectionId, errorMessage, headers);
         }
 
-        private static ConnectionDataMessage CreateConnectionDataMessage(ref MessagePackReader reader)
+        private static ConnectionDataMessage CreateConnectionDataMessage(ref MessagePackReader reader, int arrayLength)
         {
             var connectionId = ReadString(ref reader, "connectionId");
+            var seqId = arrayLength > 3 ? ReadInt32(ref reader, "sequenceId") : 0;
```
Comment (vicancy) [inline src/Microsoft.Azure.SignalR.Protocols/ServiceProtocol.cs]: `-1` as default?

File: src/Microsoft.Azure.SignalR.Protocols/ConnectionMessage.cs [hunk 3]
```diff
@@ -136,5 +136,10 @@ public ConnectionDataMessage(string connectionId, ReadOnlySequence<byte> payload
         /// Gets or sets the binary payload.
         /// </summary>
         public ReadOnlySequence<byte> Payload { get; set; }
```
Comment (vicancy) [inline src/Microsoft.Azure.SignalR.Protocols/ConnectionMessage.cs]: please revert `submodule` updates

File: src/Microsoft.Azure.SignalR.Protocols/ConnectionMessage.cs (hunk 1 above)
Comment (KKhurin) [inline src/Microsoft.Azure.SignalR.Protocols/ConnectionMessage.cs]: long? Or we're ready to deal with wrapping around max int?

Comment (terencefan) [reply src/Microsoft.Azure.SignalR.Protocols/ConnectionMessage.cs]: I don't think a client connection would hold over 0.2b messages in its lifecycle?

Comment (davidfowl) [reply src/Microsoft.Azure.SignalR.Protocols/ConnectionMessage.cs]: It should be a long

Comment (davidfowl) [reply src/Microsoft.Azure.SignalR.Protocols/ConnectionMessage.cs]: Also the property should be SequenceId

Comment (KKhurin) [reply src/Microsoft.Azure.SignalR.Protocols/ConnectionMessage.cs]: 2bln messages is just ~20 days with ~1K messages/second. Might be not that common but not unrealistic at all.

**Pull request closed without merging**